stages/org.osbuild.insights-client.config

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/python3
+
+import os
+import sys
+
+import osbuild.api
+
+
+def main(tree, options):
+    insights_client_config = options.get("config", {})
+    proxy = insights_client_config.get("proxy")
+    path = options.get("path", "/etc/insights-client/insights-client.conf")
+    path = os.path.join(tree, path.lstrip("/"))
+
+    changes = {}
+    if proxy is not None:
+        changes["proxy"] = {"key": "proxy", "value": proxy}
+
+    if not os.path.exists(path):
+        with open(path, 'w', encoding="utf8"):
+            pass
+
+    # For each of the configured options, find the first non-commented out instance
+    # of the option and replace it (if necessary). If it does not already exist, append
+    # the option to the end of the file.
+    # Keys are case insensitive, values are not. Try to preserve the key and default to
+    # camel-case.
+    with open(path, 'r', encoding='utf8') as f:
+        lines = f.readlines()
+
+    with open(path, 'w', encoding='utf8') as f:
+        for line in lines:
+            line_list = line.split('=')
+            if len(line_list) == 2:
+                key, current_value = line_list
+                entry = changes.pop(key.lower(), None)
+                if entry is not None and current_value != entry['value']:
+                    f.write(f"{key}={entry['value']}\n")
+                    continue
+            f.write(line)
```

Comment:
If I'm reading this right, this will also change values on commented lines.  We should straight-up write back any line that starts with `#`, as the comment above the read block suggests.

Reply:
I might be missing something, but it shouldn't change the values for commented lines, since the key for commented lines would be "#proxy" instead of "proxy". `entry` would be None for commented lines and so no write happens.

Reply:
Thanks for the explanation.  I was just going off the fact that the previous version was skipping lines starting with `#`.  Now those lines will be processed, but wont match any key so they'll essentially just pass through.